pkg/certificate/service.go

```diff
@@ -71,10 +74,58 @@ func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, erro
 	var localFileKeyPEM []byte
 	var localFileIdentity *InstanceIdentity
 
+	// validate file
+	isValidFile := func(file_path string) error {
+		dir_path := filepath.Dir(file_path)
+		var target_path string
+		_, file_err := os.Stat(file_path)
+		_, dir_err := os.Stat(dir_path)
+
+		if file_err == nil {
+			// validate file path
+			target_path = file_path
+		} else if os.IsNotExist(file_err) && dir_err == nil {
+			// validate dir path
+			target_path = dir_path
+		} else {
+			return fmt.Errorf("file path not exist: %w, %w", file_err, dir_err)
+		}
+
+		err := unix.Access(target_path, unix.W_OK)
+		if err != nil {
+			return fmt.Errorf("file permission error: %w", err)
+		}
+
+		return nil
+	}
```

Comment:
First of all, writing validation method under the `pkg/certificate/service.go` does not look cool to me. Write it under `/pkg/util`, maybe with a file name `validate-file-path.go`